torch/_inductor/graph.py

```diff
@@ -673,28 +681,26 @@ def run_node(self, n: torch.fx.Node):
             # with infallible strides.
             # 2: as_strided ops, we need make sure its input has same size/stride with
             # eager model to align with eager behavior.
-            as_strided_ops = [
-                torch.ops.aten.as_strided.default,
-                torch.ops.aten.as_strided_.default,
-                torch.ops.aten.as_strided_scatter.default,
-            ]
             if any(
                 user.op == "output" or user.target in as_strided_ops for user in n.users
             ) and isinstance(n.meta["val"], torch.Tensor):
-                strides = n.meta["val"].stride()
-                dense = torch._prims_common.is_non_overlapping_and_dense(n.meta["val"])
                 # requiring a stride order for a non-dense output wouldn't
                 # recreate the same strides, and would fail with view, defer for now.
+                strides = n.meta["val"].stride()
+                dense = torch._prims_common.is_non_overlapping_and_dense(n.meta["val"])
                 if dense and len(strides):
                     stride_order = ir.get_stride_order(strides)
-
-                    if (
-                        len(result.get_size()) == 4
-                        and n in self.nodes_prefer_channels_last
-                        and n.name not in self.user_visible_outputs
-                    ):
-                        stride_order = ir.NHWC_STRIDE_ORDER
```

Comment:
@shunting314, I don't know why converting the result's stride only for one of node's users is output op or as_strided_op, I think we shouldn't have such a condition, please correct me.

Reply:
I think the point here is, in general we don't need force any stride for a node unless we return it. This way, the compiler can decide whatever stride is best it thinks when handling that node

Reply:
I am confused about why converting a node results to channels last here(it should be out of this condition), there has a condition that one of the node users is graph output rather than return this node.